test-chainid/chainid-dependent-behaviour.js

```diff
@@ -0,0 +1,93 @@
+/* globals artifacts */
+import chai from "chai";
+import bnChai from "bn-chai";
+
+import { setupENSRegistrar } from "../helpers/upgradable-contracts";
+import { setupColonyNetwork, setupMetaColonyWithLockedCLNYToken, giveUserCLNYTokensAndStake } from "../helpers/test-data-generator";
+import { forwardTime, getActiveRepCycle, advanceMiningCycleNoContest, getValidEntryNumber, checkErrorRevert } from "../helpers/test-helper";
+
+import { MINING_CYCLE_DURATION, DEFAULT_STAKE, SUBMITTER_ONLY_WINDOW } from "../helpers/constants";
+
+const { expect } = chai;
+const ENSRegistry = artifacts.require("ENSRegistry");
+const ChainId = artifacts.require("ChainId");
+
+chai.use(bnChai(web3.utils.BN));
+
+contract("Contract Storage", (accounts) => {
+  const MINER1 = accounts[5];
+
+  let metaColony;
+  let clnyToken;
+  let colonyNetwork;
+  let chainId;
+
+  before(async () => {
+    const cid = await ChainId.new();
```

Comment:
I'm looking slanty-eyed at these variable names. Calling the contract `MultiChain` would help.

Reply:
Yeah, doing:

```
const multiChain = await MultiChain.new()
chainId = await multiChain.getChainId()
```

Would be much better.